stratum/hal/lib/phal/tai/tai_switch_configurator.cc

```diff
@@ -20,16 +22,34 @@ namespace tai {
 // TaiSwitchConfigurator::Make method makes an instance of
 // TaiSwitchConfigurator
 ::util::StatusOr<std::unique_ptr<TaiSwitchConfigurator>>
-TaiSwitchConfigurator::Make() {
-  return absl::WrapUnique(new TaiSwitchConfigurator());
+TaiSwitchConfigurator::Make(TaiInterface* tai_interface) {
+  return absl::WrapUnique(new TaiSwitchConfigurator(tai_interface));
 }
 
 // TaiSwitchConfigurator::CreateDefaultConfig method generates a default
 // config using the TAI API.
+// Note that we stores TAI object id in the "vendor_specific_id" field since
+// we need to use that ID later.
 ::util::Status TaiSwitchConfigurator::CreateDefaultConfig(
     PhalInitConfig* phal_config) const {
-  auto optical_card = phal_config->add_optical_cards();
-  optical_card->set_slot(1);
+  ASSIGN_OR_RETURN(auto modules, tai_interface_->GetModuleIds());
+  uint32 module_index = 1;
```

Comment:
I'm. not sure about this part. But it can probably improved later.
Why do we need increasing module indices and the module id form TAI?

Reply:
I think we keep as is and open a P3 investigation issue.